src/System.Memory/src/System/Buffers/ReadOnlySequence_helpers.cs

```diff
@@ -16,25 +16,14 @@ namespace System.Buffers
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal bool TryGetBuffer(in SequencePosition start, in SequencePosition end, out ReadOnlyMemory<T> data, out SequencePosition next)
         {
-            SequenceType type;
-            int startIndex = 0;
-            int endIndex = 0;
             next = default;
-            object startObject = start.GetObject();
-            if (startObject != null)
+            var startObject = start.GetObject();
+            if (startObject == null)
             {
-                GetTypeAndIndices(start.GetInteger(), end.GetInteger(), out type, out startIndex, out endIndex);
-            }
-            else
-            {
-                type = SequenceType.Empty;
+                data = default;
+                return false;
             }
-
-            int length = endIndex - startIndex;
-            object endObject = end.GetObject();
-
-            if (type != SequenceType.MultiSegment && type != SequenceType.Empty && startObject != endObject)
-                ThrowHelper.ThrowInvalidOperationException_EndPositionNotReached();
+            GetTypeAndIndices(start.GetInteger(), end.GetInteger(), out var type, out var startIndex, out var endIndex);
```

Comment:
nit: don't use var here. explicitly state the type. only use var where you can statically infer what the type is (for instance if the right hand side is `new ...`.